File: pdlearning/src/microservices/course-microservice/Api/Application/RequestDtos/ContentRequest/ChangeContentOrderRequest.cs
using System;
using Microservice.Course.Application.Models;

namespace Microservice.Course.Application.RequestDtos
{
    public class ChangeContentOrderRequest
    {
        public Guid Id { get; set; }

        public Guid CourseId { get; set; }

        public Guid? ClassRunId { get; set; }

        public MovementDirection Direction { get; set; }

        public MovementContentType Type { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/dtos/change-content-order-request.ts
import { CourseContentItemType } from '../models/course-content-item.model';

export interface IChangeContentOrderRequest {
  id: string;
  direction: MovementDirection;
  type: CourseContentItemType;
  courseId: string;
  classRunId?: string;
}

export enum MovementDirection {
  Up = 'Up',
  Down = 'Down'
}
